pkg/pinecone/config/seed/component.json

```diff
@@ -0,0 +1,99 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Pinecone Connector Component Spec",
+  "type": "object",
+  "properties": {
+    "task": {
+      "title": "Task",
+      "enum": [
+        "QUERY",
+        "UPSERT"
+      ],
+      "default": "QUERY"
+    }
+  },
+  "allOf": [
+    {
+      "if": {
+        "properties": {
+          "task": {
+            "const": "QUERY"
+          }
+        }
+      },
+      "then": {
+        "type": "object",
+        "required": [
+          "topK"
+        ],
+        "properties": {
+          "namespace": {
+            "type": "string",
+            "description": "The namespace to query"
+          },
+          "topK": {
+            "type": "integer",
```

Comment:
Please use `string` type for all fields in component json-schema for now.
Because we will use template language `{{ xxx.yy.zz }}` and inline-codeblock syntax `{ 0 }`, `{ true }` for the component setting in our first version, they are all string.

Reply:
updated